### PR TITLE
Reimplement "Save which contact model caused an infection (#42)".

### PR DESCRIPTION
**Pull request by tobiasraabe**

### CODE REVIEW

File: src/sid/simulate.py [hunk 1]
```diff
@@ -83,13 +83,13 @@ def get_simulate_func(
...
             "testing_states", "countdowns", "contacts", "countdown_draws", "group_codes"
-            "infection_reason" and "other".
+            and "other".
         optional_state_columns (dict): Dictionary with categories of state columns
             that can additionally be added to the states DataFrame, either for use in
             contact models and policies or to be saved. Most types of columns are added
             by default, but some of them are costly to add and thus only added when
             needed. Columns that are not in the state but specified in ``saved_columns``
-            will not be saved. The categories are "contacts" and "reason_for_infection".
+            will not be saved. The sole category is currently "contacts".
```
Comment (janosg) [inline src/sid/simulate.py:92]: I think this is outdated and there are more columns. For example in `testing_demand.py` you use:

image

Comment (janosg) [reply src/sid/simulate.py:92]: Even though I came up with this concept I think I would get rid of the `optional_state_columns` and instead keep a list (in `config.py`) of columns that are only generated if they are in the `saved_columns`. It is just not an intuitive interface.

**Review by janosg — APPROVED** [review]
Very nice. Could you add a short section in `states.rst` that documents the new columns?

File: src/sid/simulate.py (hunk 1 above)
Comment (tobiasraabe) [reply src/sid/simulate.py:92]: Postponed and planned in #54.

**Pull request merged**